src/Illuminate/Http/Client/RequestException.php

```diff
@@ -26,6 +33,37 @@ public function __construct(Response $response)
         $this->response = $response;
     }
 
+    /**
+     * Enable truncation of the exception message.
+     *
+     * @return void
+     */
+    public static function truncate()
+    {
+        static::$truncateAt = 120;
+    }
+
+    /**
+     * Disable truncation of the exception message.
+     *
+     * @return void
+     */
+    public static function dontTruncate()
+    {
+        static::$truncateAt = false;
+    }
+
+    /**
+     * Set the truncation length for the exception message.
+     *
+     * @param  int  $length
+     * @return void
+     */
+    public static function truncateAt($length)
+    {
+        static::$truncateAt = $length;
+    }
```

Comment:
Is it intentional, that `truncateAt()` can take _anything_ as a parameter because it is only type-hinted in the PHPDoc? If that is true, wouldn't it make sense, to call `truncateAt()` from `truncate()` and `dontTruncate()` so that, if the variable name changes, you only have to change it in one other place?

Reply:
This is the case with various setters in Laravel and first party packages. I also didn't want to apply a parameter or property type as the existing property in the class doesn't have a type. Happy to change it if the team wants it changed 👍

Reply:
> the existing property in the class doesn't have a type.

Didn't you add `$truncateAt` yourself? From the changed files, it looks like it. Or did you mean the other properties of the class?

> This is the case with various setters in Laravel and first party packages.

Afaik, this is mostly for historical reasons coming from a time when PHP didn't have (much) native typing. But the codebase is improved towards typing, so I'd say, newly introduced functionality tends to have types when possible.

Reply:
> Didn't you add $truncateAt yourself? From the changed files, it looks like it. Or did you mean the other properties of the class?

The other properties of the class

> Afaik, this is mostly for historical reasons coming from a time when PHP didn't have (much) native typing. But the codebase is improved towards typing, so I'd say, newly introduced functionality tends to have types when possible.

I think this is the case with new classes but not new properties on existing classes that already have properties with undefined types. Otherwise there'd be a mishmash of some properties with and without types.

Reply:
This mismatch would be in looks only but not in functionality